### ISSUE TITLE
Test files for DM4 IO seem to be DM3 files

### ISSUE THREAD

**Issue by cmeyer** [hyperspy/hyperspy#2282]
Notice: the `0x00000003` as the 1st 4 bytes of the file. DM4 files will have `0x00000004` as the 1st 4 bytes.

```
(RELEASE_next_minor)$ hexdump -C hyperspy/tests/io/dm4_2D_data/test-11.dm4 | head
00000000  00 00 00 03 00 00 5f b4  00 00 00 01 01 00 00 00  |......_.........|
00000010  00 0e 14 00 12 44 6f 63  75 6d 65 6e 74 4f 62 6a  |.....DocumentObj|
00000020  65 63 74 4c 69 73 74 00  00 00 00 00 01 14 00 00  |ectList.........|
00000030  01 00 00 00 00 12 14 00  13 41 6e 6e 6f 74 61 74  |.........Annotat|
00000040  69 6f 6e 47 72 6f 75 70  4c 69 73 74 00 00 00 00  |ionGroupList....|
00000050  00 00 15 00 0e 41 6e 6e  6f 74 61 74 69 6f 6e 54  |.....AnnotationT|
00000060  79 70 65 25 25 25 25 00  00 00 01 00 00 00 03 14  |ype%%%%.........|
00000070  00 00 00 15 00 0f 42 61  63 6b 67 72 6f 75 6e 64  |......Background|
00000080  43 6f 6c 6f 72 25 25 25  25 00 00 00 09 00 00 00  |Color%%%%.......|
00000090  0f 00 00 00 00 00 00 00  03 00 00 00 00 00 00 00  |................|
```

This may have implications for the correctness of the DM4 reader and writer. I did not do further testing.

**Comment by ericpre** [comment]
Thanks for the bug report, indeed it seems there was only one correct dm4 file (EELS_SI.dm4).
This should be fixed in #2282. See PR #2282 for detailed explanation.

The dm4 reader seems to be fine and hyperspy does not write dm4 file.

**Comment by francisco-dlp** [comment]
Fixed in #2285
